modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,204 @@
+        :param body: The string representation of the HTTP request body.
+        :param query_params: A string representing the HTTP request query parameters.
+        :return: A tuple containing two dict objects:
+            The first dict contains the HTTP headers that are used to calculate the HTTP signature.
+            The second dict contains the HTTP headers that must be added to the outbound HTTP request.
+        """
+
+        if body is None:
+            body = ''
+        else:
+            body = json.dumps(body)
+
+        # Build the '(request-target)' HTTP signature parameter.
+        target_host = urlparse(self.configuration.host).netloc
+        target_path = urlparse(self.configuration.host).path
+        request_target = method.lower() + " " + target_path + resource_path
+        if query_params:
+            raw_query = urlencode(query_params).replace('+', '%20')
+            request_target += "?" + raw_query
+
+        # Get current time and generate RFC 1123 (HTTP/1.1) date/time string.
+        now = datetime.datetime.now()
+        stamp = mktime(now.timetuple())
+        cdate = formatdate(timeval=stamp, localtime=False, usegmt=True)
+        created = now.strftime("%s")
+        if self.configuration.signature_max_validity is not None:
+            expires = (now + self.configuration.signature_max_validity).strftime("%s")
+
+        signed_headers_dict = {}
+        request_headers_dict = {}
+        for hdr_key in self.configuration.signed_headers_dict:
+            hdr_key = hdr_key.lower()
+            if hdr_key == '(request-target)':
+                value = request_target
+            elif hdr_key == '(created)':
+                value = created
+            elif hdr_key == '(expires)':
+                value = expires
+            elif hdr_key == 'date':
+                value = cdate
+                request_headers_dict['Date'] = '{0}'.format(cdate)
+            elif hdr_key == 'digest':
+                request_body = body.encode()
+                body_digest, digest_prefix = self.get_message_digest(request_body)
+                b64_body_digest = b64encode(body_digest.digest())
+                value = digest_prefix + b64_body_digest.decode('ascii')
+                request_headers_dict['Digest'] = '{0}{1}'.format(digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == 'host':
+                value = target_host
+                request_headers_dict['Host'] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers_dict[hdr_key] = value
+
+        # If the user has not provided any signed_headers, the default must be set to '(created)'.
+        if len(self.configuration.signed_headers_dict) == 0:
+            signed_headers_dict['(created)'] = created
+
+        return signed_header_dict, request_headers_dict
+
+    def get_http_signature_headers(self, resource_path, method, headers, body, query_params):
```

Comment:
Why not just pass in configuration here?
Then you can delete the parent class and make these all functions.

Reply:
done.